### ISSUE TITLE
`audit_year` mismatches in API-returned data

### ISSUE THREAD

**Issue by stucka** [GSA-TTS/FAC#2316]
### Describe the bug

The API is returning faulty `audit_year` numbers, which may be a legitimate problem or an artifact of test data.

For example, UEIs with records in both the API and the CSV, comparing reported fiscal years and reported fiscal year end dates:
```
H7GPYDN4LVJ9: 2022 vs. 2022 ... 2023-01-31 vs. 31-JAN-22
P2TDH1JCJD31: 2022 vs. 2022 ... 2023-06-30 vs. 30-JUN-22
D6ELHXJ2GPZ9: 2022 vs. 2022 ... 2023-06-30 vs. 30-JUN-22
MYQRKM53KGA9: 2022 vs. 2022 ... 2023-06-30 vs. 30-JUN-22
U7X5E477A4S5: 2022 vs. 2022 ... 2023-06-30 vs. 30-JUN-22
F2F1MB6M6K39: 2022 vs. 2022 ... 2023-04-30 vs. 30-APR-22
TAN6F9EAQ7V3: 2022 vs. 2022 ... 2023-06-30 vs. 30-JUN-22
N8TKG3QGKD85: 2022 vs. 2022 ... 2023-06-30 vs. 30-JUN-22
NAVHCCU1KD44: 2022 vs. 2022 ... 2023-06-30 vs. 30-JUN-22
MC64LDFG2716: 2022 vs. 2022 ... 2023-07-31 vs. 30-JUN-22
WZZVLPLH4LX3: 2022 vs. 2022 ... 2023-03-31 vs. 31-MAR-22
YH6GPJPGDMN6: 2022 vs. 2022 ... 2023-06-30 vs. 30-JUN-22
FM3USR82YYT4: 2022 vs. 2022 ... 2023-06-30 vs. 30-JUN-22
```

### Steps to reproduce the bug

where requestheaders contains an API key and "general-2.csv" is a newish copy of the Census Bureau's general file extracted from the ZIP:

```
r = requests.get("https://api.fac.gov/general?limit=1000&date_created=gte.2023-09-23", headers=requestheaders)
badaudityears = {}
for entry in r.json():
    uei = entry['auditee_uei']
    badaudityears[uei] = {
        "api_audit_year": entry['audit_year'],
        "csv_audit_year": None,
        "api_fy_end_date": entry["fy_end_date"],
        "csv_fy_end_date": None,
    } 

# Find CSV matches
highlights = []
with open("general-2.csv", "r", encoding="utf-8") as infile:
    reader = csv.DictReader(infile)
    for row in reader:
        if row['UEI'] in badaudityears:
            highlights.append(row)

for row in highlights:
    uei = row['UEI']
    badaudityears[uei]['csv_audit_year'] = row['AUDITYEAR']
    badaudityears[uei]['csv_fy_end_date'] = row['FYENDDATE']

for uei in badaudityears:
    if badaudityears[uei]['csv_audit_year']:      # Drop entries with no UEI matches
        entry = badaudityears[uei]
        print(f"{uei}: {entry['api_audit_year']} vs. {entry['csv_audit_year']} ... {entry['api_fy_end_date']} vs. {entry['csv_fy_end_date']}")
```

### Expected Behavior

`audit_year` variables should be a year newer when the data is a year newer.

### Code of Conduct

- [X] I agree to abide by the [Digital.gov Community Guidelines](https://digital.gov/communities/community-guidelines/) and the [TTS Code of Conduct](https://handbook.tts.gsa.gov/code-of-conduct/). Respect your peers, use plain language, and be patient.
- [X] I checked the [current issues](https://github.com/GSA-TTS/FAC/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by jadudm** [comment]
Hi @stucka ,

At some point, we convinced ourselves that we should disseminate the audit year based on the start of the auditee's fiscal year as opposed to the end. I know we were careful in this decision at the time, but clearly made the wrong call.

Friday, we began a small epic to track and do the work to correct this error. Fortunately, this does not impact the data taken in, and we will put a fix into place promptly. We'll track the work here:

https://github.com/GSA-TTS/FAC/issues/2323

First, thank you for an excellent report. It made it completely obvious to us what was going on.

And, second, thank you for engaging with the new API right away.
